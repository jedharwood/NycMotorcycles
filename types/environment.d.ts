declare global {
  namespace NodeJS {
    interface ProcessEnv {
      NEXT_PUBLIC_API_MOCKING: string;
      SELECTOR_PRODUCT: string;
      SELECTOR_PRODUCT_PRICE: string;
      SELECTOR_PRODUCT_TIME: string;
      SELECTOR_PRODUCT_TITLE: string;
      SELECTOR_PRODUCT_BID: string;
      YAHOO_AUCTION_PROFILE_PAGE_URL: string;
      MAILGUN_API_KEY: string;
      MAILGUN_DOMAIN: string;
      DELIVERY_EMAIL: string;
    }
  }
}

export {};
