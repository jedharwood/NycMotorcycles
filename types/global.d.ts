type TimeRemaining = {
  time: number | undefined;
  unit: 'days' | 'hours' | 'minutes';
};

type ActiveAuction = {
  title?: string;
  url?: string;
  image: GridImage;
  bidders?: string;
  timeRemaining?: TimeRemaining;
  currentPrice?: string;
  promptDecisionPrice?: string;
};

type GridImage = {
  imageSrc: string;
  imageAlt: string;
  width?: number;
  height?: number;
  onImageClick?: () => void;
};

type ArchiveBikeKey = keyof typeof archive;

type ContactFormData = {
  email: string;
  senderName: string;
  subject: string;
  message: string;
};

type InputFieldProps = {
  type: 'email' | 'text' | 'text-area';
  placeholder: string;
  name: 'email' | 'senderName' | 'subject' | 'message';
  register: UseFormRegister<ContactFormData>;
  error: FieldError | undefined;
  label: string;
};

type ComponentColour = 'green' | 'red';

type SoldOrCall = 'sold' | 'call';

type TheNumbersOneToFour = 1 | 2 | 3 | 4;

type NavLinkDetails = {
  href: string;
  text: string;
};

type FooterProps = {
  address: string;
  disclaimer: string;
  chevronText: string;
  altTextInstagramButton: string;
};

type ChildElementPosition = 'under-title' | 'bottom';

type TextDisplayProps = {
  title?: string;
  textContent?: string[];
  childElement?: JSX.Element;
  childElementPosition?: ChildElementPosition;
  isOpaque?: boolean;
  hasBorder?: boolean;
  textContentCentred?: boolean;
  borderColour?: ComponentColour;
  subTitle?: string;
  footer?: string;
};