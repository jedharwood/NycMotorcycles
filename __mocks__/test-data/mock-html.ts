export const mockHtml = `
    <!DOCTYPE html>
    <html lang="ja">
    <head>
    </head>
    <body id="allContents">
        <div class="Products__list">
            <ul class="Products__items">
                <li class="Product">
                    <div class="Product__image"> <a
                            data-auction-title="Auction 1"
                            href="https://page.auctions.yahoo.co.jp/jp/auction/1">
                            <img 
                                class="Product__imageData" 
                                src="https://images.auctions.yahoo.co.jp/image/1" 
                                width="133" 
                                height="100" 
                                alt="Auction 1"
                            >
                        </a>
                    </div>
                    <div class="Product__detail">
                        <p class="Product__title">
                            <a class="Product__titleLink js-rapid-override js-browseHistory-add"
                            href="https://page.auctions.yahoo.co.jp/jp/auction/1">
                                Auction 1
                            </a>
                        </p>
                        <div class="Product__priceInfo">
                            <div class="Product__price">
                                <dt class="Product__label">現在</dt>
                                <dd class="Product__priceValue gv-u-colorOrange5">30,000,000円</dd>
                            </div>
                            <div class="Product__price">
                                <dt class="Product__label">即決</dt>
                                <dd class="Product__priceValue">33,000,000円</dd>
                            </div>
                        </div>
                        <div class="Product__otherInfo">
                            <dl class="Product__bidWrap">
                                <dt class="Product__label">入札</dt>
                                <dd class="Product__bid">0</dd>
                            </dl>
                            <dl class="Product__timeWrap">
                                <dt class="Product__label">残り</dt>
                                <dd class="Product__time">7日</dd>
                            </dl>
                        </div>
                    </div>
                </li>
                <li class="Product">
                    <div class="Product__image"> <a
                            data-auction-title="Auction 2"
                            href="https://page.auctions.yahoo.co.jp/jp/auction/2">
                            <img 
                                class="Product__imageData" 
                                src="https://images.auctions.yahoo.co.jp/image/2" 
                                width="134" 
                                height="89" 
                                alt="Auction 2"
                            >
                        </a>
                    </div>
                    <div class="Product__detail">
                        <p class="Product__title">
                            <a class="Product__titleLink js-rapid-override js-browseHistory-add"
                            href="https://page.auctions.yahoo.co.jp/jp/auction/2">
                                Auction 2
                            </a>
                        </p>
                        <div class="Product__priceInfo">
                            <div class="Product__price">
                                <dt class="Product__label">即決</dt>
                                <dd class="Product__priceValue gv-u-colorOrange5">25,000円</dd>
                            </div>
                        </div>
                        <div class="Product__otherInfo">
                            <dl class="Product__bidWrap">
                                <dt class="Product__label">入札</dt>
                                <dd class="Product__bid">0</dd>
                            </dl>
                            <dl class="Product__timeWrap">
                                <dt class="Product__label">残り</dt>
                                <dd class="Product__time">4時間</dd>
                            </dl>
                        </div>
                    </div>
                </li>
                <li class="Product">
                    <div class="Product__image"> <a
                            data-auction-title="Auction 3"
                            href="https://page.auctions.yahoo.co.jp/jp/auction/2">
                        </a>
                    </div>
                    <div class="Product__detail">
                        <div class="Product__priceInfo">
                            <div class="Product__price">
                                <dt class="Product__label">即決</dt>
                            </div>
                        </div>
                        <div class="Product__otherInfo">
                            <dl class="Product__bidWrap">
                                <dt class="Product__label">入札</dt>
                            </dl>
                        </div>
                        <dl class="Product__timeWrap">
                            <dt class="Product__label">残り</dt>
                            <dd class="Product__time">4分</dd>
                        </dl>
                    </div>
                </li>
                <li class="Product">
                    <div class="Product__image"> <a
                            data-auction-title="Auction ４"
                            href="https://page.auctions.yahoo.co.jp/jp/auction/2">
                        </a>
                    </div>
                </li>
            </ul>
        </div>
    </div>				
    </body>
    </html>
`;