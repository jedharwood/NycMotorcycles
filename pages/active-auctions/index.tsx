import { AuctionCard } from '@/components/auction-card'
import { FC, useEffect, useState } from 'react'
import { yahooAuctionProfilePage } from '../../utilities/resource-utilities'
import { TextDisplay } from '@/components/text-display'
import { Spinner } from '@/components/spinner'
import { LinkButton } from '@/components/link-button'
import { HeadElement } from '@/components/head-element'

const ActiveAuctionPage: FC = (): JSX.Element => {
  const [activeAuctions, setActiveAuctions] = useState<ActiveAuction[]>([])
  const [isLoading, setIsLoading] = useState<boolean>(true)

  useEffect(() => {
    const fetchData = async () => {
      const response = await fetch('http://localhost:3000/api/auction-scraper')
      const data = await response.json()
      setActiveAuctions(data.activeAuctions)
      setIsLoading(false)
    }
    fetchData()
  }, [])

  const auctionDisplay = (): JSX.Element => {
    if (isLoading) return <></>

    const yahooAuctionLinkButton: JSX.Element = (
      <LinkButton
        text="pg.active-auctions.yahoo-auctions-button"
        href={yahooAuctionProfilePage}
        type="anchor"
      />
    )

    return !activeAuctions.length ? (
      <TextDisplay
        title="pg.active-auctions.no-auctions.title"
        textContent={['pg.active-auctions.no-auctions.body']}
        childElement={yahooAuctionLinkButton}
      />
    ) : (
      <ul className="space-y-6">
        {activeAuctions.map((auction) => {
          return (
            <li key={auction.title}>
              <AuctionCard {...auction} />
            </li>
          )
        })}
      </ul>
    )
  }

  return (
    <>
      <HeadElement
        pageTitle="pg.active-auctions.head.meta.title"
        content="pg.active-auctions.head.meta.content"
      />
      <main>
        <Spinner loading={isLoading} />
        {auctionDisplay()}
      </main>
    </>
  )
}

export default ActiveAuctionPage
