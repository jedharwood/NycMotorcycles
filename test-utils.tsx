import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import en from './languages/en.json';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import { QueryClient, QueryClientProvider } from 'react-query';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient();

  return (
    <MemoryRouterProvider>
      <QueryClientProvider client={queryClient}>
        <IntlProvider locale='en' messages={en}>
          {children}
        </IntlProvider>
      </QueryClientProvider>
    </MemoryRouterProvider>
  );
};

const renderWithProviders = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { renderWithProviders as render };
