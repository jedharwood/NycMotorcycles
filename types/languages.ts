import en from '../languages/en.json';
import ja from '../languages/ja.json';

export const langs = {
    en: 'en',
    ja: 'ja',
};

type Messages = {
    [key: string]: {
        [key: string]: string;
    };
}; 

export const messages: Messages = {
    en,
    ja,
};