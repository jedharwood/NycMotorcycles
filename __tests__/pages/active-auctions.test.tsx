import { useQuery } from 'react-query';

import { activeAuctions } from '@/__mocks__/test-data/active-auctions';
import ActiveAuctionsPage from '@/pages/active-auctions';
import { render, screen } from '@/test-utils';
import { langs } from '@/types/languages';

jest.mock('react-query', () => ({
    ...jest.requireActual('react-query'),
    useQuery: jest.fn(),
}));

afterEach(() => {
    jest.clearAllMocks();
});

// mock fetch as well as useQuery/instead of...
describe('ActiveAuctionsPage', () => {
    it('should render spinner when data is loading', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: true,
        });
        render(<ActiveAuctionsPage />);

        const spinner: HTMLElement = await screen.findByTestId(
            'loading-spinner',
            {},
            { timeout: 2000 }
        );
        expect(spinner).toBeInTheDocument();
    });

    it('should render access denied display in English when status is 403', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 403,
                data: {},
            },
        });
        const { container } = render(<ActiveAuctionsPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render access denied display in Japanese when status is 403', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 403,
                data: {},
            },
        });
        const { container } = render(<ActiveAuctionsPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });

    it('should render there has been a problem display in English when status is 500', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 500,
                data: {},
            },
        });
        const { container } = render(<ActiveAuctionsPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render there has been a problem display in Japanese when status is 500', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 500,
                data: {},
            },
        });
        const { container } = render(<ActiveAuctionsPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });

    it('should render no active auctions display in English when activeAuctions is empty array', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 200,
                data: {
                    activeAuctions: []
                },
            },
        });
        const { container } = render(<ActiveAuctionsPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render no active auctions display in Japanese when activeAuctions is empty array', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 200,
                data: {
                    activeAuctions: []
                },
            },
        });
        const { container } = render(<ActiveAuctionsPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });

    it('should render active auctions in English', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 200,
                data: {
                    activeAuctions
                },
            },
        });
        const { container } = render(<ActiveAuctionsPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render active auctions in Japanese', async () => {
        (useQuery as jest.Mock).mockReturnValue({
            isLoading: false,
            data: {
                status: 200,
                data: {
                    activeAuctions
                },
            },
        });
        const { container } = render(<ActiveAuctionsPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });
});
