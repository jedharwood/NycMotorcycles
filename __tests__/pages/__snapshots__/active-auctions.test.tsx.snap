// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveAuctionsPage should render access denied display in English when status is 403 1`] = `
<div>
  <main>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          There has been a problem connecting you to Yahoo auctions
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          If you are accessing the site from the UK/EU try connecting through a VPN.
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          href="https://auctions.yahoo.co.jp/seller/lazzamoto?"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              Yahoo auction page
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render access denied display in Japanese when status is 403 1`] = `
<div>
  <main>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          placeholder
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          placeholder
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          href="https://auctions.yahoo.co.jp/seller/lazzamoto?"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              placeholder
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render active auctions in English 1`] = `
<div>
  <main>
    <div
      class="grid gap-6 lg:grid-cols-2"
    >
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="An image of a cat"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vincent Black Shadow
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Current price
                </td>
                <td>
                  10,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Prompt decision price
                </td>
                <td>
                  15,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Time remaining
                </td>
                <td>
                  56 minutes
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Bidders
                </td>
                <td>
                  3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Another image of a cat"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Another Vincent Black Shadow
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Current price
                </td>
                <td>
                  98,560円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Time remaining
                </td>
                <td>
                  9 hours
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Bidders
                </td>
                <td>
                  3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Are you starting to see a pattern forming, yet?"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Yet Another Vincent Black Shadow
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Current price
                </td>
                <td>
                  33,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Prompt decision price
                </td>
                <td>
                  35,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Time remaining
                </td>
                <td>
                  3 days
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Bidders
                </td>
                <td>
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Another image of a cat"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Another Vincent Black Shadow with a much longer title, this time
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Current price
                </td>
                <td>
                  98,560円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Time remaining
                </td>
                <td>
                  9 hours
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  Bidders
                </td>
                <td>
                  3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
    </div>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render active auctions in Japanese 1`] = `
<div>
  <main>
    <div
      class="grid gap-6 lg:grid-cols-2"
    >
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="An image of a cat"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vincent Black Shadow
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  10,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  15,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  56 分
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Another image of a cat"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Another Vincent Black Shadow
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  98,560円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  9 時間
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Are you starting to see a pattern forming, yet?"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Yet Another Vincent Black Shadow
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  33,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  35,000円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  3 日
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article
        class="card-hover grid w-full gap-4 rounded-md bg-stone-600 bg-opacity-90 py-4 px-6 text-stone-50 hover:bg-opacity-100 sm:grid-cols-3"
      >
        <a
          class="relative h-fit rounded-md bg-gray-500 sm:col-span-1"
          href="https://www.google.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Another image of a cat"
            class="w-full rounded-md hover:opacity-70"
            src="/api/image-proxy?url=http%3A%2F%2Fplacekitten.com%2F200%2F300"
          />
        </a>
        <div
          class="sm:col-span-2"
        >
          <a
            class="text-xl font-medium hover:underline hover:opacity-100 md:opacity-80"
            href="https://www.google.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Another Vincent Black Shadow with a much longer title, this time
          </a>
          <table
            class="mt-4 w-full table-auto"
          >
            <tbody>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  98,560円
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  9 時間
                </td>
              </tr>
              <tr
                class="text-sm hover:underline sm:text-lg"
              >
                <td
                  class="pr-4 sm:opacity-80"
                >
                  placeholder
                </td>
                <td>
                  3
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
    </div>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render no active auctions display in English when activeAuctions is empty array 1`] = `
<div>
  <main>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          There are currently no active auctions
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          Click the button below to visit out Yahoo auction page. Follow the Yahoo auction account for updates when new items are listed.
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          href="https://auctions.yahoo.co.jp/seller/lazzamoto?"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              Yahoo auction page
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render no active auctions display in Japanese when activeAuctions is empty array 1`] = `
<div>
  <main>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          placeholder
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          placeholder
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          href="https://auctions.yahoo.co.jp/seller/lazzamoto?"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              placeholder
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render there has been a problem display in English when status is 500 1`] = `
<div>
  <main>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          There has been a problem connecting you to Yahoo auctions
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          Please contact the site for support.
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          href="/contact"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              Contact
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;

exports[`ActiveAuctionsPage should render there has been a problem display in Japanese when status is 500 1`] = `
<div>
  <main>
    <section
      class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
    >
      <div
        class="text-center "
      >
        <h2
          class="text-xl font-medium opacity-80 md:text-2xl"
        >
          placeholder
        </h2>
      </div>
      <div
        class="space-y-4 undefined"
      >
        <p>
          placeholder
        </p>
      </div>
      <div
        class="flex justify-center"
      >
        <a
          href="/contact"
        >
          <div
            class="flex justify-center"
          >
            <button
              class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
              type="button"
            >
              placeholder
            </button>
          </div>
        </a>
      </div>
    </section>
  </main>
</div>
`;
